spec.html

```diff
@@ -39904,6 +39904,57 @@ <h1 id="ao-issuperreference">IsSuperReference ( _V_ )</h1>
         1. Return ? ForIn/OfBodyEvaluation(|BindingIdentifier|, |Statement|, _keyResult_, ~enumerate~, ~varBinding~, _labelSet_).
       </emu-alg>
     </emu-annex>
+
+    <emu-annex id="sec-html-document-dot-all-exotic-object">
+      <h1>HTML Document Dot All Exotic Object</h1>
+      <p>An <dfn>HTML document dot all exotic object</dfn> is an exotic object that behaves like undefined in the <emu-xref href="#sec-toboolean">ToBoolean</emu-xref> and <emu-xref href="#sec-abstract-equality-comparison">Abstract Equality Comparison</emu-xref> abstract operations and when used as an operand for the <emu-xref href="#sec-typeof-operator">typeof operator</emu-xref>. HTML document dot all exotic objects have all the ordinary object internal methods and slots.</p>
+      <emu-note>
+        <p>These objects are never created by this specification. However, the <a href="https://html.spec.whatwg.org/multipage/obsolete.html#dom-document-all"><code>document.all</code> object</a> in web browsers is a host-created HTML document dot all exotic object that exists for web compatibility purposes. There are no other known examples of this type of object.</p>
+      </emu-note>
+
+      <emu-annex id="sec-html-document-dot-all-exotic-object-to-boolean">
+        <h1>Changes to ToBoolean</h1>
+        <p>The result column in <emu-xref href="#table-10"></emu-xref> for an argument type of Object is replaced with the following algorithm:</p>
+        <emu-alg>
+          1. If _argument_ is an HTML document dot all exotic object, return *true*.
+          1. Return *false*.
+        </emu-alg>
+      </emu-annex>
+
+      <emu-annex id="sec-html-document-dot-all-exotic-object-aec">
+        <h1>Changes to Abstract Equality Comparison</h1>
+        <p>The following steps are inserted after step 3 of the <emu-xref href="#sec-abstract-equality-comparison">Abstract Equality Comparison</emu-xref> algorithm:</p>
+        <emu-alg>
+          1. If _x_ is an undefined-like exotic object and _y_ is either *null* or *undefined*, return *true*.
+          1. If _x_ is either *null* or *undefined* and _y_ is an undefined-like exotic object, return *true*.
+        </emu-alg>
+      </emu-annex>
+
+      <emu-annex id="sec-html-document-dot-all-exotic-object-typeof">
+        <h1>Changes to the typeof operator</h1>
+        <p>The following table entry is inserted into <emu-xref href="#table-35"></emu-xref> immediately preceeding the entry for "Object (implements [[Call]])":</p>
+        <emu-table caption="Additional Typeof Table Entries">
+          <table>
+            <tr>
+              <th>
+                Type of _val_
+              </th>
+              <th>
+                Result
+              </th>
+            </tr>
+            <tr>
+              <td>
+                Object (HTML dot all exotic object)
```

Comment:
xref? I guess we're already in the section due to annex B funtimes, so maybe it doesn't need to be.